src/common/util_bug.h

```diff
@@ -120,17 +148,17 @@
 #define IF_BUG_ONCE__(cond,var)                                         \
   if (( {                                                               \
       static int var = 0;                                               \
-      int bool_result = (cond);                                         \
```

Comment:
Is it because the new branch prediction macros are only enabled in unittests, and this function is only used in regular code?

So we prefer to disable branch prediction than do it in a warning-suppressing way?

Could we perhaps mention the fact that these new macros are used only for unittests explicitly in their documentation?

Reply:
We've moved the branch prediction from this macro into the IF_BUG_ONCE() definition below, where we add ASSERT_PREDICT_UNLIKELY_()

Reply:
The new ASSERT_PREDICT() stuff is designed to only do branch prediction in the release-build case.  In the unit-test-build case, it's designed to give the warnings on the conditional case.  

I've added another commit, with a comment to explain that.